File: Examples/VBasic/GroupDocs.Search.Examples.VBasic/Utilities.vb
Imports GroupDocs.Search.Events

'ExStart:commonutilitiesclass
Public Class Utilities

    Public Const indexPath As String = "../../../../Data/Documents Indexes/"
    Public Const indexPath2 As String = "../../../../Data/Documents Indexes2/"
    Public Const mergeIndexPath1 As String = "../../../../Data/Index Merging/Index1/"
    Public Const mergeIndexPath2 As String = "../../../../Data/Index Merging/Index2/"
    Public Const mainMergedIndexesPath As String = "../../../../Data/Index Merging/Main Merged Indexes/"
    Public Const documentsPath As String = "../../../../Data/Documents/"
    Public Const documentsPath2 As String = "../../../../Data/Documents2/"
    Public Const documentsPath3 As String = "../../../../Data/Documents3/"
    Public Const synonymFilePath As String = "../../../../Data/synonyms.txt"
    Public Const stopWordsFilePath As String = "../../../../Data/MyStopWords.txt"
    Public Const exportedStopWordsFilePath As String = "../../../../Data/MyExportedStopWords.txt"
    Public Const mySynonymFilePath As String = "../../../../Data/mySynonyms.txt"
    Public Const licensePath As String = "D:/Aspose Projects/License/GroupDocs.Total.lic"
    Public Const pathToPstFile As String = "D:/MyOutlookDataFile.pst"
    Public Const pathToPasswordProtectedFile As String = "../../../../Data/Documents/Password Protected Document.docx"
    Public Const pathToPasswordProtectedFile2  As String= "../../../../Data/Documents/Password Protected Document2.docx"
    Public Const pathToPasswordProtectedFile3 As String = "../../../../Data/Documents/Password Protected Document3.docx"
    Public Const exportedHomophonesFilePath As String = "../../../../Data/MyExportedHomophones.txt"
    Public Const homophonesFilePath As String = "../../../../Data/MyHomophones.txt"
    Public Const aliasFilePath As String = "../../../../Data/MyAliases.txt"
    Public Const exportedAliasFilePath As String = "../../../../Data/MyExportedAliases.txt"
    Public Const spellingDictionaryFilePath As String = "../../../../Data/MySpellingDictionary.txt"
    Public Const exportedSpellingDictionaryFilePath As String = "../../../../Data/MyExportedSpellingDictionary.txt"

    ''' <summary>
    ''' Apply license 
    ''' </summary>
    Public Shared Sub ApplyLicense()
        'initialize License
        Dim lic As New License()
        'Set license
        lic.SetLicense(licensePath)
    End Sub
    ''' <summary>
    ''' Index operation finished
    ''' </summary>
    ''' <param name="sender">Object</param>
    ''' <param name="e">OperationFinishedArg</param>
    Public Shared Sub index_OperationFinished(sender As Object, e As OperationFinishedArg)
        Dim time As DateTime = e.Time
        ' Time when documents added
        Dim indexId As Guid = e.IndexId
        ' Index Id
        Dim indexFolder As String = e.IndexFolder
        ' Index Folder
        Dim status As IndexStatus = e.Status
        ' Index Status
        Dim operationType As OperationType = e.OperationType
        ' Operation Type.
    End Sub
    ''' <summary>
    ''' Index operation finished
    ''' </summary>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Shared Function index_OperationFinished() As Object
        Throw New NotImplementedException
    End Function
End Class
'ExEnd:commonutilitiesclass

File: Examples/CSharp/GroupDocs.Search.Examples.CSharp/Utilities.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using GroupDocs.Search;
using GroupDocs.Search.Events;

namespace GroupDocs.Search_for_.NET
{
    //ExStart:commonutilitiesclass
    public class Utilities
    {
        public const string indexPath = "../../../../Data/Documents Indexes/";
        public const string indexPath2 = "../../../../Data/Documents Indexes2/";
        public const string mergeIndexPath1 = "../../../../Data/Index Merging/Index1/";
        public const string mergeIndexPath2 = "../../../../Data/Index Merging/Index2/";
        public const string mainMergedIndexesPath = "../../../../Data/Index Merging/Main Merged Indexes/";
        public const string documentsPath = "../../../../Data/Documents/";
        public const string documentsPath2 = "../../../../Data/Documents2/";
        public const string documentsPath3 = "../../../../Data/Documents3/";
        public const string synonymFilePath = "../../../../Data/synonyms.txt";
        public const string stopWordsFilePath = "../../../../Data/MyStopWords.txt";
        public const string exportedStopWordsFilePath = "../../../../Data/MyExportedStopWords.txt";
        public const string mySynonymFilePath = "../../../../Data/mySynonyms.txt";
        public const string licensePath = "D:/Aspose Projects/License/GroupDocs.Total.lic";
        public const string pathToPstFile = "D:/MyOutlookDataFile.pst";
        public const string pathToPasswordProtectedFile =  "../../../../Data/Documents/Password Protected Document.docx";
        public const string pathToPasswordProtectedFile2 = "../../../../Data/Documents/Password Protected Document2.docx";
        public const string pathToPasswordProtectedFile3 = "../../../../Data/Documents/Password Protected Document3.docx";
        public const string exportedHomophonesFilePath = "../../../../Data/MyExportedHomophones.txt";
        public const string homophonesFilePath = "../../../../Data/MyHomophones.txt";
        public const string aliasFilePath = "../../../../Data/MyAliases.txt";
        public const string exportedAliasFilePath = "../../../../Data/MyExportedAliases.txt";
        public const string spellingDictionaryFilePath = "../../../../Data/MySpellingDictionary.txt";
        public const string exportedSpellingDictionaryFilePath = "../../../../Data/MyExportedSpellingDictionary.txt";
        public const string oldIndexFolderPath = "../../../../Data/Old index";
        public const string alphabetFilePath = "../../../../Data/MyAlphabet.txt";
        public const string exportedAlphabetFilePath = "../../../../Data/MyExportedAlphabet.txt";
        public const string publicKey = "[Your Dynabic.Metered public key]";
        public const string privateKey = "[Your Dynabic.Metered private key]";



        /// <summary>
        /// Apply license 
        /// </summary>
        public static void ApplyLicense()
        {
            //initialize License
            License lic = new License();
            //Set license
            lic.SetLicense(licensePath);
        }

        /// <summary>
        /// Allows using Dynabic.Metered account to run library in licensed mode
        /// Feature is supported for version 17.06 or greater
        /// </summary>
        public static void UseDynabicMeteredAccount() {
            //ExStart:UseDynabicMeteredAccount
            // initialize Metered API and set-up credentials
            new Metered().SetMeteredKey(publicKey, privateKey);
            // do indexing and searching in licensed mode 
            //ExEnd:UseDynabicMeteredAccount
        }

        /// <summary>
        /// Index operation finished
        /// </summary>
        /// <param name="sender">Object</param>
        /// <param name="e">OperationFinishedArg</param>
        public static void index_OperationFinished(object sender, OperationFinishedArg e)
        {
            DateTime time = e.Time; // Time when documents added
            Guid indexId = e.IndexId; // Index Id
            string indexFolder = e.IndexFolder; // Index Folder
            IndexStatus status = e.Status; // Index Status
            OperationType operationType = e.OperationType; // Operation Type.
        }

        /// <summary>
        /// Index operation finished
        /// </summary>
        /// <param name="sender">Object</param>
        /// <param name="e">BaseIndexArg</param>
        internal static void index_OperationFinished(object sender, BaseIndexArg e)
        {
            throw new NotImplementedException();
        }
    }
    //ExEnd:commonutilitiesclass
}
